kernel/common/hash.rb

```diff
@@ -632,6 +632,9 @@ def [](key)
     end
   end
 
+  # Used internally to get around subclasses redefining #[]
+  alias_method :__get__, :[]
+
```

Comment:
There's no need to document this anywhere in particular as the statement applies to the Ruby code as a whole (including any existing code). We'll simply inform people about this whenever they run into any problems.